obal/data/modules/check_koji_build.py

```diff
@@ -23,23 +23,35 @@ def main():
     package = module.params['package']
     koji_executable = module.params['koji_executable']
 
+    command = ['buildinfo', nevr]
     try:
-        if tag:
-            command = ['latest-build', '--quiet', tag, package]
-            build = koji(command, koji_executable)
-            build = build.split(' ')[0]
+        output = koji(command, koji_executable)
+    except KojiCommandError as error:
+        module.fail_json(changed=False, msg=error.message, command=error.command)
 
-            module.exit_json(changed=False, tagged_version=build, exists=(nevr == build))
-        else:
-            command = ['buildinfo', nevr]
-            output = koji(command, koji_executable)
-            exists = 'No such build' not in output
+    exists = "BUILD: %s" % (nevr) in output
 
-            module.exit_json(changed=False, exists=exists)
+    if tag:
+        exists_for_tag = False
 
-    except KojiCommandError as error:
-        module.fail_json(changed=False, msg=error.message, command=error.command)
+        for line in output.split("\n"):
+            if line.startswith("Tags:"):
+                exists_for_tag = tag in line
```

Comment:
while probably uncommon, this would match if we configure `tag` as `tag1` but the build is tagged into `tag1-candidate`

IMHO more correct:
```python
if line.startswith("Tags: "):
    tags = line.split()[1:]
    exists_for_tag = tag in tags
```